lib/df-detection.c

```diff
@@ -64,17 +61,27 @@ void end_system_call(void)
 		current->addresses = NULL;
 	}
 }
-void print_pairs(void)
+void report(void)
 {
 	int i;
-	for (i = 0; i < current->df_index; i++) {
-		pr_err("First %px len %lu \nSecond %px len %lu\n",
-		       current->pairs[i].first->start_address,
-		       current->pairs[i].first->len,
-		       current->pairs[i].second->start_address,
-		       current->pairs[i].second->len);
+	pr_err("BUG: Intersection Detected \n ");
+	pr_err("==================================================================\n");
+	if (panic_on_warn) {
+		pr_err("System Call Number: %ld\n", current->syscall_num);
+		for (i = 0; i < current->df_index; i++) {
+			pr_err("First %px len %lu Caller %ps \nSecond %px len "
+			       "%lu Caller %ps \n",
+			       current->pairs[i].first->start_address,
+			       current->pairs[i].first->len,
+			       current->pairs[i].first->caller,
+			       current->pairs[i].second->start_address,
+			       current->pairs[i].second->len,
+			       current->pairs[i].second->caller);
+		}
+		pr_err("==================================================================\n");
```

Comment:
is it panic_on_warn not enough ? i know that we discussed to add panic to the function report but i thought that because panic_on_warn wasn't work but now i fix what i did wrong and it printing all the data we need in syzkaller report

Reply:
Yes, panic_on_warn is also enough, but it should only be checked before calling panic().
The report must be printed unconditionally, i.e. if you start your kernel without panic_on_warn, you should see a bunch of reports in the log, but if you set panic_on_warn then you'll crash after the first one.